github.cake

```diff
@@ -0,0 +1,78 @@
+#addin "nuget:https://api.nuget.org/v3/index.json?package=Octokit&version=0.48.0"
+#addin "nuget:https://api.nuget.org/v3/index.json?package=NuGet.Versioning&version=5.7.0"
+#addin "nuget:https://api.nuget.org/v3/index.json?package=Cake.Yaml&version=3.1.1"
+
+using Octokit;
+using NuGet.Versioning;
+
+static CakeGitHubReleaseInfo _cakeGitHubReleaseInfo;
+
+public static CakeGitHubReleaseInfo GetCakeGitHubReleaseInfo(this ICakeContext context, DirectoryPath releaseDir)
+{
+    if (!(_cakeGitHubReleaseInfo is null))
+    {
+        return _cakeGitHubReleaseInfo;
+    }
+
+    // We cache the latest Cake release information in a file to speed up local development and also
+    // to decrease the chance of getting rate limited by GitHub when running unauthenticated
```

Comment:
Instead of caching we can put the reading of GitHub release information into a separate task, which is not called from the `preview` target, but a dependency of the `GetArtifacts` tasks, so that it is called in the `Build` / `Default` target. This is how it is already done with reading of Cake source code and extension packages.

Reply:
I'm fine with reading the GitHub release information in a separate task, but we still need to get the release version info during the Preview Task as well in order to see the banner in the docs, and in the getting started, thus the Preview task would also have to depend on this new Task and this is where caching is important given the Preview task can be run a number of times during local dev. I remember hitting the limit a couple of times yesterday - I believe the limit is 30 unauthenticated requests.

Reply:
If the task writes the result to a file, like the caching implementation currently does, it would work. It would then be in the responsibility of the developer to call the `GetArtifacts` task once to get the release information, otherwise you won't have this information. This is how it currently is also implemented for extensions or Cake Source. If I don't call `GetArtifacts` before calling `Preview` I won't have this information in the built website (and in most of the cases this is even the better, because faster option).

Reply:
Got it. It seems that the end goal is that the `Preview` task would not access any external resource and just build using local resources. So in preview we would have to either use a fallback value for these variables or display empty (e.g. in the [getting started markdown page](https://github.com/cake-build/website/pull/1462/commits/f428886e85c628d50f11d07ffd78f693138baee5#diff-4e129ca53399e95e0d8764573e9142df4c72e3c7cd70a7d4ddda1cb565c8e7df))